### ISSUE TITLE
`prefer_const_literals_to_create_immutables` support for extension types

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4722]
Related to #4720, 

```dart
extension type E(List<int> i) { }

var e = E([1]);
```

should we treat `E` as immutable and produce a [`prefer_const_literals_to_create_immutables`](https://dart.dev/tools/linter-rules/prefer_const_literals_to_create_immutables) lint on `[1]`?

/fyi @goderbauer @a14n @eernstg

**Comment by bwilkerson** [comment]
The documentation for the lint doesn't make it clear what the motivation for the lint is, and I don't remember it. Could you describe the motivation? It's hard to know how to answer the question without understanding what the lint is trying to accomplish.

**Comment by lrhn** [comment]
Presumably, the `@immutable` annotations is intended to convey that the class instances should be deeply immutable.
By passing in only deeply immutable arguments to the constructor, that is more likely to stay true, and the only way to ensure that  arguments are deeply immutable is that they are constant expressions. (Or, presumably, instances of other `@immutable`-annotated classes.)

The documentation for `@immutable` isn't that clear. I'm guessing it shows *intent* to be deeply immutable, but the only direct effect it has on the annotated class is to require all fields to be final. This lint is trying to force you towards full immutability.

**Comment by pq** [comment]
> Presumably, the @immutable annotations is intended to convey that the class instances should be deeply immutable.

I think this is right.

> This lint is trying to force you towards full immutability.

And this.

**Comment by pq** [comment]
Fixed w/ https://github.com/dart-lang/sdk/commit/039640fad17e980373fdeb4edfc5c145bac85784.
